include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -980,6 +980,45 @@ SpatialOperatorBase<dim, Number>::prescribe_initial_conditions(VectorType & velo
   pressure = pressure_double;
 }
 
+template<int dim, typename Number>
+void
+SpatialOperatorBase<dim, Number>::interpolate_analytical_solution(VectorType & velocity,
+                                                                  VectorType & pressure,
+                                                                  double const time) const
+{
+  AssertThrow(field_functions->analytical_solution_velocity,
+              dealii::ExcMessage("FieldFunctions::analytical_solution_velocity not set"));
+  AssertThrow(field_functions->analytical_solution_pressure,
+              dealii::ExcMessage("FieldFunctions::analytical_solution_pressure not set"));
+
+  velocity.zero_out_ghost_values();
+  pressure.zero_out_ghost_values();
```

Comment:
It is not clear to me why we have these lines of code here, while we do not have them e.g. in the function `prescribe_initial_conditions()` above.

Reply:
I will move the line to the analytical time integrator:)

Reply:
I am fine with this. I think a comment explaining why we have/need these lines of code would help me to better understand the code (e.g.: what would happen if one forgot to write these lines of code? segmentation-fault or performance penalty or ...?).

We can also do this as a follow-up so that we can merge the present PR soon.

Reply:
`dealii::VT::interpolate()` simply throws an exception with zeroed out ghost values. During time stepping the vectors are not zeroed out though, so we have to care about that ourselves.